### PR TITLE
UIU-3132 - Reading Room table sorting in edit screen.

### PR DESCRIPTION
**Pull request by manvendra-s-rathore**
[UIU-3132](https://folio-org.atlassian.net/browse/UIU-3132) - Reading Room table sorting in User Edit screen.

## Screencast
chrome_bGrLjAUYdb

## Test Coverage
I do not see any further scope of increasing test case coverage in EditReadingRoomAccess.js file.

Also the coverage of getFormatter.js is 100% in local coverage report. It is not reflecting in SonarCloud Coverage Report.
image

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by manvendra-s-rathore** [comment]
Hey @folio-org/fe-tl-reviewers , can someone please look into this PR.

File: src/components/EditSections/EditReadingRoomAccess/EditReadingRoomAccess.test.js
Comment (Terala-Priyanka) [inline src/components/EditSections/EditReadingRoomAccess/EditReadingRoomAccess.test.js:1]: Could you add a unit test for sorting columns after editing a record, and assert the record position?

File: src/components/EditSections/EditReadingRoomAccess/EditReadingRoomAccess.js [hunk 1]
```diff
@@ -47,13 +76,24 @@ const EditReadingRoomAccess = ({
...
-        contentData={formData}
+        contentData={sortedRecordsDetails.data}
         columnMapping={columnMapping}
         visibleColumns={visibleColumns}
         formatter={getFormatter(form)}
         columnWidths={columnWidths}
+        sortOrder={sortedRecordsDetails.order}
+        sortDirection={`${sortedRecordsDetails.direction}ending`}
+        onHeaderClick={(e, meta) => setSortedRecordsDetails(
+          getReadingRoomSortedData(e, meta, sortedRecordsDetails)
```
Comment (Terala-Priyanka) [inline src/components/EditSections/EditReadingRoomAccess/EditReadingRoomAccess.js:94]: add semicolon add the end of the statement. Doesn't lint complain about this?

File: src/components/EditSections/EditReadingRoomAccess/getFormatter.js [hunk 2]
```diff
@@ -9,17 +9,18 @@ import { Select, TextArea } from '@folio/stripes/components';
 import { rraColumns, READING_ROOM_ACCESS_OPTIONS } from './constants';
 
 export const getFormatter = (form) => {
-  const updateRecord = (record, val, name, rowIndex) => {
-    const fieldState = form.getFieldState('readingRoomsAccessList');
-    if (fieldState?.value && fieldState.value[rowIndex]) {
-      form.change(`readingRoomsAccessList[${rowIndex}][${name}]`, val);
+  const updateRecord = (record, inputValue, name) => {
+    const rraFormList = form.getFieldState('readingRoomsAccessList')?.value;
```
Comment (Terala-Priyanka) [inline src/components/EditSections/EditReadingRoomAccess/getFormatter.js:13]: does "rraList" sound better?

File: src/components/EditSections/EditReadingRoomAccess/getFormatter.js [hunk 3]
```diff
@@ -9,17 +9,18 @@ import { Select, TextArea } from '@folio/stripes/components';
...
+    const recordIndexInRRAList = rraFormList?.findIndex((field) => field?.readingRoomId === record?.readingRoomId);
```
Comment (Terala-Priyanka) [inline src/components/EditSections/EditReadingRoomAccess/getFormatter.js:14]: does "recordIndex" suffice instead of "reacordIndexInRRAList"?

**Review by Terala-Priyanka — CHANGES_REQUESTED** [review]
Please address the comments.

File: src/components/UserDetailSections/ReadingRoomAccess/ReadingRoomAccess.js [hunk 4]
```diff
@@ -134,7 +121,9 @@ const ReadingRoomAccess = (props) => {
         formatter={getFormatter(lastUpdatedDetails)}
         sortOrder={sortedRecordsDetails.order}
         sortDirection={`${sortedRecordsDetails.direction}ending`}
-        onHeaderClick={onSort}
+        onHeaderClick={(e, meta) => setSortedRecordsDetails(
+          getReadingRoomSortedData(e, meta, sortedRecordsDetails)
```
[the remark this answers is not in the document]
Comment (manvendra-s-rathore) [reply src/components/UserDetailSections/ReadingRoomAccess/ReadingRoomAccess.js:125]: It is not required here as **getReadingRoomSortedData()** will return the sortedRecords which will set the state of **sortedRecordsDetails**.

File: src/components/EditSections/EditReadingRoomAccess/EditReadingRoomAccess.js (hunk 1 above)
Comment (manvendra-s-rathore) [reply src/components/EditSections/EditReadingRoomAccess/EditReadingRoomAccess.js:94]: It is not required here as getReadingRoomSortedData() will return the sortedRecords which will set the state of sortedRecordsDetails.

File: src/routes/UserRecordContainer.js
Comment (Terala-Priyanka) [inline src/routes/UserRecordContainer.js:1]: Please fix lint issues here

File: src/components/EditSections/EditReadingRoomAccess/EditReadingRoomAccess.test.js [hunk 5]
```diff
@@ -145,4 +146,27 @@ describe('EditReadingRoomAccess', () => {
     fireEvent.change(accessSelectField, { target: { value: 'NOT_ALLOWED' } });
     await waitFor(() => expect(props.form.change).toHaveBeenCalled());
   });
+
+  it('should sort the records by default by room name', () => {
+    renderEditReadingRoomAccess(props);
+    expect(document.querySelectorAll('[class^="mclCell"]')[1].innerHTML).toBe('abc');
+  });
+
+  it('should sort the records by room name when name column header is clicked', () => {
```
Comment (Terala-Priyanka) [inline src/components/EditSections/EditReadingRoomAccess/EditReadingRoomAccess.test.js:155]: can you update the title of the test to indicate that the records are sorted in desc order of reading room name?

**Review by Terala-Priyanka — APPROVED** [review]

**Comment by manvendra-s-rathore** [comment]
Hey @folio-org/fe-tl-reviewers , I am merging this PR as it is in code-review since long. Any review comments can be covered in a follow-up PR.

**Pull request merged**